core/src/hash_table/pair_meta.rs

```diff
@@ -154,4 +157,31 @@ pub mod tests {
     fn source() {
         assert_eq!(test_pair_meta().source(), test_keys().node_id());
     }
+
+    #[test]
+    /// test that we can sort pair metas with cmp
+    fn cmp() {
+        let p1 = test_pair_a();
+        let p2 = test_pair_b();
+
+        // basic ordering
+        let m_1ax = PairMeta::new(&test_keys(), &p1, "a", "x");
+        let m_1ay = PairMeta::new(&test_keys(), &p1, "a", "y");
+        let m_1bx = PairMeta::new(&test_keys(), &p1, "b", "x");
+        let m_2ax = PairMeta::new(&test_keys(), &p2, "a", "x");
+
+        // sort by pair key
+        assert_eq!(Ordering::Less, m_1ax.cmp(&m_2ax));
+        assert_eq!(Ordering::Equal, m_1ax.cmp(&m_1ax));
+        assert_eq!(Ordering::Greater, m_2ax.cmp(&m_1ax));
+        assert_eq!(Ordering::Less, m_1ay.cmp(&m_2ax));
+
+        // sort by attribute key
+        assert_eq!(Ordering::Less, m_1ax.cmp(&m_1bx));
+        assert_eq!(Ordering::Greater, m_1bx.cmp(&m_1ax));
+
+        // sort by attribute value
+        assert_eq!(Ordering::Less, m_1ax.cmp(&m_1ay));
+        assert_eq!(Ordering::Greater, m_1ay.cmp(&m_1ax));
```

Comment:
Wouldn't it be better to assert for the actual condition? So e.g.
```rust
assert!(m_1ax < m_2ax);
assert_eq!(m_1ax, m_1ax);
assert!(m_2ax > m_1ax);
assert!(m_1ay < m_2ax);
```

Reply:
well asserting equality is testing a different thing, as for the others i'm not sure

it makes sense to me that i'm testing `cmp` but i could add the operators as well to cover both